gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/service/NACKMessageService.java

```diff
@@ -0,0 +1,42 @@
+package uk.nhs.adaptors.pss.translator.service;
+
+import static uk.nhs.adaptors.pss.translator.util.DateFormatUtil.toHl7Format;
+import static uk.nhs.adaptors.pss.translator.util.template.TemplateUtil.fillTemplate;
+import static uk.nhs.adaptors.pss.translator.util.template.TemplateUtil.loadTemplate;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.github.mustachejava.Mustache;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.common.util.DateUtils;
+import uk.nhs.adaptors.pss.translator.model.NACKMessageData;
+import uk.nhs.adaptors.pss.translator.util.template.parameter.ApplicationAcknowledgmentParams;
+
+@Service
+@Slf4j
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+public class NACKMessageService {
```

Comment:
If you went with leaving only one mustache template, you could also have only one service (called for example `ApplicationAcknowledgmentMessageService`). Then you could also have only one MessageData class (but the `nackCode` couldn't be annotated with @NonNull).